### PR TITLE
Decouple fileupload and createTemporaryMessage from the global store module

### PR DESCRIPTION
**Pull request by PVince81**
1. Decouple fileUpload service from the global store
    
Moved the logic of "processFiles" into the existing "initialiseUpload" action of the fileUploadStore

2. Move createTemporaryMessage to a store action
    
Since createTemporaryMessage heavily depends on a store, moved it into a messagesStore action to decouple it from the global store module.    

Note: even though it's synchronous, the dispatch() function returns a promise so we need to treat it as if it was async.

This is prerequisite to be able to properly mock parts of the (non-global) store in https://github.com/nextcloud/spreed/pull/5522

Tested:
- [x] drag and drop upload of multiple files
- [x] upload dialog with multiple files
- [x] share existing file to the conversation from the file picker

### CODE REVIEW

File: src/store/messagesStore.js [hunk 1]
```diff
@@ -286,6 +288,64 @@ const actions = {
...
+	createTemporaryMessage(context, { text, token, uploadId, index, file, localUrl }) {
+		const messageToBeReplied = context.getters.getMessageToBeReplied(token)
+		const date = new Date()
+		let tempId = 'temp-' + date.getTime()
+		const messageParameters = {}
+		if (file) {
+			tempId += '-' + uploadId + '-' + Math.random()
+			messageParameters.file = {
+				'type': 'file',
+				'file': file,
```
Comment (nickvergessen) [inline src/store/messagesStore.js:313]:
```suggestion
				file,
```

File: src/store/messagesStore.js [hunk 2]
```diff
@@ -286,6 +288,64 @@ const actions = {
...
+			id: tempId,
+			actorId: context.getters.getActorId(),
+			actorType: context.getters.getActorType(),
+			actorDisplayName: context.getters.getDisplayName(),
+			timestamp: 0,
+			systemMessage: '',
+			messageType: '',
+			message: text,
+			messageParameters,
+			token: token,
```
Comment (nickvergessen) [inline src/store/messagesStore.js:333]:
```suggestion
			token,
```

File: src/store/messagesStore.js (hunk 1 above)
Comment (PVince81) [reply src/store/messagesStore.js:313]: this is just copy pasted from the old copy, but probably a good idea to adjust. I'll adjust them all

File: src/store/messagesStore.js [hunk 3]
```diff
@@ -286,6 +288,64 @@ const actions = {
...
+	createTemporaryMessage(context, { text, token, uploadId, index, file, localUrl }) {
+		const messageToBeReplied = context.getters.getMessageToBeReplied(token)
+		const date = new Date()
+		let tempId = 'temp-' + date.getTime()
+		const messageParameters = {}
+		if (file) {
+			tempId += '-' + uploadId + '-' + Math.random()
+			messageParameters.file = {
+				type: 'file',
+				file: file,
```
Comment (nickvergessen) [inline src/store/messagesStore.js:313]:
```suggestion
				file,
```

**Review by nickvergessen — APPROVED** [review]
Seems to still work

**Pull request merged**